crates/aranya-daemon/src/api.rs

```
@@ -364,20 +375,49 @@ impl DaemonApi for Api {
         team: api::TeamId,
         cfg: api::TeamConfig,
     ) -> api::Result<()> {
-        todo!()
+        if let Some(cfg) = cfg.quic_sync {
+            let mut guard = self.quic.lock().await;
+            let quic_data = guard.as_mut().context("quic syncing is not enabled")?;
+
+            let seed = QuicSyncSeed::<CS>::from_bytes(cfg.seed())?;
+            insert_seed(&mut quic_data.engine, &mut quic_data.store, seed.clone())
+                .context("could not insert seed into keystore")?;
+            write_seed_id(&quic_data.seed_id_path, &team, &seed.id())
+                .await
+                .context("could not write seed id to file")?;
+
+            let psk = seed.gen_psk()?;
+
+            let identity = psk.identity();
+            let secret = psk.raw_secret();
+            let psk = PresharedKey::external(identity, secret)
+                .context("unable to create PSK")?
+                .with_hash_alg(HashAlgorithm::SHA384)
+                .expect("Valid hash algorithm");
+
+            quic_data
+                .psk_send
+                .send(Msg::Insert((team, Arc::new(psk))))?;
+        }
+
+        // TODO: Implement for other syncer types
```

We should return an error if `add_team` is called without a quic syncer configured. Otherwise, this could silently fail.

---

I did leave a `todo!()` which would panic originally but I refactored it based on some [feedback](https://github.com/aranya-project/aranya/pull/258/files/6d0addfa2028122b5e57438edc8c0f0229ddd566#r2130054562) and removed the panic. I hadn't considered the possibility of having multiple syncers configured at once and they may each require different behavior.

I think it's fine to leave for now since it can't fail silently now since users can't actually start a daemon without the QUIC syncer

---

I'll just return an error for now since we only have a single syncer type. It can be refactored once we have other options